dask/dataframe/tests/test_dataframe.py

```diff
@@ -306,19 +306,110 @@ def test_describe(method, test_values):
     assert_eq(df.describe(), ddf.describe(split_every=2, percentiles_method=method))
 
 
-@pytest.mark.parametrize('method', ['tdigest', 'dask'])
-def test_describe_empty(method):
-    if method == 'tdigest':
-        pytest.importorskip('crick')
-    # https://github.com/dask/dask/issues/2326
-    ddf = dd.from_pandas(pd.DataFrame({"A": ['a', 'b']}), 2)
-    with pytest.raises(ValueError) as rec:
-        ddf.describe(percentiles_method=method)
-    assert 'DataFrame contains only non-numeric data.' in str(rec)
+@pytest.mark.parametrize('include,exclude,percentiles,subset', [
+    (None, None, None, ['c','d']),  # numeric
+    (None, None, None, ['c', 'd', 'f']),  # numeric + timedelta
+    (None, None, None, ['c', 'd', 'g']),  # numeric + bool
+    (None, None, None, ['c', 'd', 'f', 'g']),  # numeric + bool + timedelta
+    (None, None, None, ['f', 'g']),  # bool + timedelta
+    ('all', None, None, None),
+    (['number'], None, [0.25, 0.5], None),
+    ([np.timedelta64], None, None, None),
+    (['number', 'object'], None, [0.25, 0.75], None),
+    (None, ['number', 'object'], None, None),
+    (['object', 'datetime', 'bool'], None, None, None)
+])
+def test_describe(include, exclude, percentiles, subset):
+    data = {
+        'a': ["aaa", "bbb", "bbb", None, None, "zzz"] * 2,
+        'c': [None, 0, 1, 2, 3, 4] * 2,
+        'd': [None, 0, 1] * 4,
+        'e': [pd.Timestamp('2017-05-09 00:00:00.006000'),
+              pd.Timestamp('2017-05-09 00:00:00.006000'),
+              pd.Timestamp('2017-05-09 07:56:23.858694'),
+              pd.Timestamp('2017-05-09 05:59:58.938999'),
+              None,
+              None] * 2,
+        'f': [np.timedelta64(3, 'D'),
+              np.timedelta64(1, 'D'),
+              None,
+              None,
+              np.timedelta64(3, 'D'),
+              np.timedelta64(1, 'D')] * 2,
+        'g': [True, False, True] * 4
+    }
+
+    # Arrange
+    df = pd.DataFrame(data)
+
+    if subset is not None:
+        df = df.loc[:, subset]
+
+    ddf = dd.from_pandas(df, 2)
+
+    # Act
+    desc_ddf = ddf.describe(include=include, exclude=exclude, percentiles=percentiles)
+    desc_df = df.describe(include=include, exclude=exclude, percentiles=percentiles)
+
+    # Assert
+    # TODO: for timedelta columns there's overflow in var function, see #4233
+    # causing std to be nan, workaround this allowing AssertionError on std sell
+    if 'f' in desc_ddf._meta:
+        with pytest.raises(AssertionError) as info:
+            assert_eq(desc_ddf, desc_df)
+            msg = "DataFrame.iloc[:, 2] are different"
```

Comment:
I don't think this code is executed, since the `with` block should exit after the AssertionError in assert_eq (unless I'm misunderstanding).

May need to move the assert outside, though personally I wouldn't worry about the error message. I also think that @jcrist's suggestion of dropping is fine too.

Reply:
@TomAugspurger yes it is wrong indent :(  I'll just drop the column then as everyone agreed on it.